test/clj/rems/api/test_catalogue.clj

```diff
@@ -19,15 +19,49 @@
       (is (some #(str/starts-with? (:resid %) "urn:") data)))))
 
 (deftest catalogue-api-security-test
-  (testing "listing without authentication"
-    (let [response (-> (request :get (str "/api/catalogue"))
-                       handler)
-          body (read-body response)]
-      (is (response-is-unauthorized? response))
-      (is (= "unauthorized" body))))
-  (testing "listing with wrong API-Key"
-    (is (= "unauthorized"
-           (-> (request :get (str "/api/catalogue"))
-               (assoc-in [:headers "x-rems-api-key"] "invalid-api-key")
-               handler
-               (read-body))))))
+  (testing "catalogue-is-public true"
+    (with-redefs [rems.config/env (assoc rems.config/env :catalogue-is-public true)]
+      (testing "with no authentification"
+        (let [response (-> (request :get "/api/catalogue")
+                           handler)]
+          (is (response-is-ok? response))))
+      (let [api-key "42"]
```

Comment:
I think there is some unnecessary `let` blocks here, you could move this under `testing` and combine the two `let` statements. Same situation repeats more after this one.

Reply:
Another option is to move the api-key and such higher and just not use them where they are not supposed to be used.

Reply:
I dont really see how that is dramatically different from the other test cases we already ahve, but can you maybe explain the nuance?

Reply:
You use the `api-key` defined here only once, and you define `api-key` again later with the same value. Some possibilities:
- just use the constant `"42"` instead of defining `api-key`
- make one common `api-key` definition that all the test cases use

It's fine like this, the code just looks a bit funny.

Reply:
The code has unnecessary nesting / repeating stuff so could be cleaned up.

Reply:
If it's unclear I can do the change and you can check the diff.

Reply:
that would be great